packages/la-boite/src/repos/local_git_repo.ts

```diff
@@ -1,10 +1,56 @@
-import { GitRepo, Reference } from './interfaces'
 import { GitProcess, IGitResult } from 'dugite'
+import { CommandBus } from 'git-en-boite-command-bus'
+import { GitRepo, Reference } from './interfaces'
 import fs from 'fs'
 
+export class Commit {
+  constructor(public readonly message: string, public readonly branchName: string) {}
+
+  static withMessage(message: string) {
+    return new Commit(message, 'master')
+  }
+
+  onBranch(branchName: string): Commit {
+    return new Commit(this.message, branchName)
+  }
+}
+
+export class Init {
+  constructor(public readonly isBare: boolean) {}
+
+  static bareRepo() {
+    return new Init(true)
+  }
+
+  static withWorkingDirectory() {
+    return new Init(false)
+  }
+}
+
+type Commands = Init | Commit
+
+const handleInit = (repo: LocalGitRepo, command: Init) =>
+  repo.git('init', ...(command.isBare ? ['--bare'] : []))
+
+const handleCommit = async (repo: LocalGitRepo, command: Commit) => {
+  const { branchName, message } = command
+  await repo.git('config', 'user.email', 'test@example.com')
+  await repo.git('config', 'user.name', 'Test User')
+  await repo.git('checkout', '-b', branchName)
+  await repo.git('commit', '--allow-empty', '-m', message)
+}
+
 export class LocalGitRepo implements GitRepo {
   path: string
 
+  static async openForCommands(path: string) {
+    const repo = await this.open(path)
+    const commandBus = new CommandBus<LocalGitRepo, Commands>(repo)
+    commandBus.handle(Init, handleInit)
+    commandBus.handle(Commit, handleCommit)
+    return commandBus.do.bind(commandBus)
+  }
+
```

Comment:
Could this be extracted away? It looks like plumbing that `LocalGitRepo` could be ignorant about.

Reply:
Ya. I felt like using it in the static factory method was as OK place to keep the plumbing for now. I have an incling to have two separate sets of commands - one broader one for testing, and a smaller (probabably overlapping) set for production commands. I was thinking it made sense to encapsulate those choices in some kind of wrapper / factory, so this is where I've put it for now.

Any thoughts on where else it could naturally go?

Reply:
`LocalGitRepoCommand(Bu)s`? Instantiate that and all the plumbing is done for you?